### DISCUSSION TITLE
Does the new router ship with Kong 3.x more efficient for dealing with a large number of regex routes as compare to previous version?

### DISCUSSION THREAD

**Discussion by LoremipsumSharp** [Kong/kong#10602] [Help]
Currently, we are using Kong 2.X to handle a large number of regex routes, about 2000 regex routes per instance, each instance experienced a  high CPU usage 
According to [this video](https://www.youtube.com/watch?v=V2xp9_oJYJ4&t=418s&ab_channel=Kong) Kong 3.x has implement a new router by radix tree, I wonder if the new router is more efficient as compare to previous version when deal with a large number of regex routes,

**Comment by 23henne** [comment]
What's your value of `nginx_http_lua_regex_cache_max_entries`?

**Reply by LoremipsumSharp** [reply]
8192

**Reply by 23henne** [reply]
OK, so it should be high enough. How many paths does each of your regex route contain?
